qiita_db/software.py

```diff
@@ -579,10 +579,27 @@ def from_file(cls, fp, update=False):
 
                 if (client_id != instance.client_id or
                         client_secret != instance.client_secret):
-                    raise qdb.exceptions.QiitaDBOperationNotPermittedError(
-                        'The (client_id, client_secret) pair of the plugin '
-                        '"%s" version "%s" does not match the one in the '
-                        'system' % (name, version))
+                    if update:
+                        sql = """INSERT INTO qiita.oauth_identifiers
```

Comment:
I think I understand your question now (thanks @antgonza ). The select that can return a lot of row is inside an `EXISTS` and it is only used to check if that row already exists or not. The first select, can only return 1 row, since the values are passes as a parameter, i.e. it is effectively doing something similar to `SELECT 42`.